runtime/metricsview/search_query.go

```diff
@@ -0,0 +1,292 @@
+		return nil, errDruidNativeSearchUnimplemented
+	}
+	var query map[string]interface{}
+	if a.Root.Where != nil {
+		// NOTE :: this does not work for measure filters.
+		// The query planner resolves them to joins instead of filters.
+		rows, err := olap.Execute(ctx, &drivers.Statement{
+			Query:            fmt.Sprintf("EXPLAIN PLAN FOR SELECT 1 FROM %s WHERE %s", *a.Root.FromTable, a.Root.Where.Expr),
+			Args:             a.Root.Where.Args,
+			DryRun:           false,
+			Priority:         0,
+			LongRunning:      false,
+			ExecutionTimeout: 0,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		if !rows.Next() {
+			return nil, fmt.Errorf("failed to parse filter")
+		}
+
+		var (
+			planRaw string
+			resRaw  string
+			attrRaw string
+		)
+		err = rows.Scan(&planRaw, &resRaw, &attrRaw)
+		if err != nil {
+			return nil, err
+		}
+
+		var plan []druid.QueryPlan
+		err = json.Unmarshal([]byte(planRaw), &plan)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(plan) == 0 {
+			return nil, fmt.Errorf("failed to parse policy filter")
+		}
+		if plan[0].Query.Filter == nil {
+			// if we failed to parse a filter we return and run UNION query.
+			// this can happen when the row filter is complex
+			// TODO: iterate over this and integrate more parts like joins and subfilter in policy filter
+			return nil, errDruidNativeSearchUnimplemented
+		}
+		query = *plan[0].Query.Filter
+	}
+	dsn, err := druid.GetDSN(olap.(drivers.Handle).Config())
+	if err != nil {
+		return nil, err
+	}
+	if dsn == "" {
+		return nil, fmt.Errorf("druid connector config not found in instance")
+	}
+
+	// Build a native query
+	limit := 100
+	if a.Root.Limit != nil {
+		limit = int(*a.Root.Limit)
+	}
+	dims := make([]string, 0)
+	virtualCols := make([]druid.NativeVirtualColumns, 0)
+	for _, f := range a.Root.DimFields {
+		dim, err := a.lookupDimension(f.Name, true)
+		if err != nil {
+			return nil, err
+		}
+		// if the dimension is a expression we need a virtual column that can be scanned in SearchDimensions
+		if dim.Expression != "" {
+			virtualCols = append(virtualCols, druid.NativeVirtualColumns{
+				Type:       "expression",
+				Name:       fmt.Sprintf("%v_virtual_native", f.Name), // The name of the virtual column should not clash with actual column
+				Expression: dim.Expression,
+			})
+			dims = append(dims, fmt.Sprintf("%v_virtual_native", f.Name))
+		} else {
+			dims = append(dims, trimQuotes(f.Expr))
+		}
+	}
+	req := druid.NewNativeSearchQueryRequest(trimQuotes(*a.Root.FromTable), q.Search, dims, virtualCols, limit, q.TimeRange.Start, q.TimeRange.End, query) // TODO: timestamps may be nil!
+
+	// Execute the native query
+	var res druid.NativeSearchQueryResponse
+	nq := druid.NewNativeQuery(druidSQLDSN.ReplaceAllString(dsn, "/v2/"))
+	err = nq.Do(ctx, req, &res, req.Context.QueryID, logger)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I think this should be pushed into the Druid driver, so we don't have DSN and similar parsing logic here. Could use a similar pattern like the DuckDB appender, for example:
```go
h := druid.NewNativeClientFromHandle(olap.(drivers.Handle))
res, err := h.Search(a.Root.FromTable, q.Search, ...)
```